### PR TITLE
Added optional field argument to  shr_string_listCreateField()

### PR DESCRIPTION
**Pull request by samsrabin**
Given integers field1 and fieldN, plus string strBase, generates a list of field names. As shr_string_listCreateField(), but not just limited to 1,numFields. (See #15.)

### CODE REVIEW

**Comment by jedwards4b** [comment]
This PR is quite old, is it still needed or desired?

**Comment by billsacks** [comment]
Reading back through the comments, it looks like all that's needed is fixing the unit tests and adding a new one for this new functionality. I'm happy to do that, but before I do, @samsrabin can you please confirm whether this is still needed?

**Comment by samsrabin** [comment]
Yes, this is used for MCT in my `rx_crop_calendars2` branch [here](https://github.com/samsrabin/CTSM/blob/029d57bd87e1f9d2d930d75e3e31e4a357d0b178/src/cpl/mct/cropcalStreamMod.F90#L134-L137). I've pushed a couple of commits fixing what you mentioned. Sorry I let this fall off the back burner and behind the stove!

File: src/shr_string_mod.F90 [hunk 1]
```diff
@@ -1765,23 +1765,32 @@ end function shr_string_listFromSuffixes
...
   !===============================================================================
-  function shr_string_listCreateField( numFields, strBase ) result ( retString )
+  function shr_string_listCreateField( fieldN, strBase, field1 ) result ( retString )
 
     implicit none
 
-    integer(SHR_KIND_IN), intent(in) :: numFields   ! number of fields
-    character(len=*)    , intent(in) :: strBase     ! input string base
-    character(SHR_KIND_CXX)          :: retString   ! colon delimited field list
+    integer(SHR_KIND_IN), intent(in), optional :: field1
```
[the remark this answers is not in the document]
Comment (samsrabin) [reply src/shr_string_mod.F90]: Fixed: 1b92e04

File: src/shr_string_mod.F90 [hunk 2]
```diff
@@ -1797,14 +1806,23 @@ function shr_string_listCreateField( numFields, strBase ) result ( retString )
     if ( debug > 1 .and. t01 < 1 ) call shr_timer_get( t01,subName )
     if ( debug > 1 ) call shr_timer_start( t01 )
 
+    if (present(field1)) then
+       l_field1 = field1
+    else
+       l_field1 = 1
+    end if
+
+    SHR_ASSERT_FL( ( l_field1 < fieldN ) , __FILE__, __LINE__)
```
[the remark this answers is not in the document]
Comment (samsrabin) [reply src/shr_string_mod.F90]: Fixed: 741e173

**Comment by billsacks** [comment]
Okay, I have gotten the unit tests passing and added a new unit test. I also merged up to (close to) the latest CESM_share master.

I'm going to run one system test with baseline comparisons just to double check this; assuming that's good, I think this is good to merge.

Moving forward, we shouldn't need to continue to develop the MCT-based streams files that this is needed for. However, given that this work was already done, and that this does make this function more generally useful, I figure we can merge this PR.

**Review by billsacks — APPROVED** [review]

**Comment by billsacks** [comment]
Thanks for merging this @jedwards4b . And I can confirm that these two tests passed and are bit-for-bit with this change:

```
ERP_D_Ld3_Vmct_PS.f09_g17.I2000Clm50Sp.cheyenne_intel.clm-prescribed
SMS_D_Ld1_P48x1_Vmct.f10_f10_mg37.I2000Clm50BgcCru.izumi_nag.clm-af_bias_v7
```

(These tests cover the MCT-based LAI stream code, which is the only CESM code that I can find that currently uses this function.)

**Pull request merged**